### PR TITLE
Improved Update Folder behaviour (Update Folder Uses Plugin Name)

### PR DESCRIPTION
**Pull request by Xemorr**
This is just a reopening of https://github.com/PaperMC/Paper/pull/6575 (to put it plainly, I'm not amazing at git and didn't want to deal with rebasing when I couldn't remember how I set my project up back in September so decided to just reimplement the patch on a fresh Paper installation).

### CODE REVIEW

File: patches/api/0377-Update-Folder-Uses-Plugin-Name.patch [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
++    private File checkUpdate(@NotNull File file) throws InvalidPluginException {
+         if (updateDirectory == null || !updateDirectory.isDirectory()) {
+-            return;
++            return file;
++        }
++        PluginLoader pluginLoader = getPluginLoader(file);
++        try {
++            String pluginName = pluginLoader.getPluginDescription(file).getName();
++            for (File updateFile : updateDirectory.listFiles()) {
++                if (updateFile.isFile()) {
```
Comment (kennytv) [inline patches/api/0377-Update-Folder-Uses-Plugin-Name.patch]: Early continues for this and the pluginname equals

**Review by kennytv — COMMENTED** [review]
Looks good otherwise

File: patches/api/0377-Update-Folder-Uses-Plugin-Name.patch (hunk 1 above)
Comment (Xemorr) [reply patches/api/0377-Update-Folder-Uses-Plugin-Name.patch]: Hopefully I understood your review correctly :) changes made

**Comment by kennytv** [comment]
aaaa prs from master branches

**Comment by Xemorr** [comment]
Is that a major issue?

**Comment by NoahvdAa** [comment]
It's not a huge problem, but it makes it a bit harder to rebase, both when merging and when you update your PR yourself (because the branch you're rebasing onto has the same name as the branch you're rebasing)

**Review by kennytv — APPROVED** [review]
Rebase plz

**Comment by yannicklamprecht** [comment]
> It's not a huge problem, but it makes it a bit harder to rebase, both when merging and when you update your PR yourself (because the branch you're rebasing onto has the same name as the branch you're rebasing)

Never had issues rebasing the same branch but different origin.


1. `git pull upstream/master --rebase`
2. Everything you need to do when doing it for a feature branch like
    1. optionally incrementing your patches number
    2. `./gradlew applyPatches`
    3. Look for conflicts while applying patches and fix them
    4. `./gradlew rebuildPatches`

**Comment by Xemorr** [comment]
I think I've rebased it and squashed all the commits

**Pull request merged**